examples/viewer.py

```diff
@@ -162,65 +179,70 @@ def __init__(self, sim_settings: Dict[str, Any]) -> None:
 
         # configure our simulator
         self.cfg: Optional[habitat_sim.simulator.Configuration] = None
-        self.sim: Optional[habitat_sim.simulator.Simulator] = None
+        self.sim: list[habitat_sim.simulator.Simulator] = None
+        self.replay_renderer_cfg: Optional[ReplayRendererConfiguration] = None
+        self.replay_renderer: Optional[BatchReplayRenderer] = None
         self.reconfigure_sim()
 
         # compute NavMesh if not already loaded by the scene.
-        if (
-            not self.sim.pathfinder.is_loaded
-            and self.cfg.sim_cfg.scene_id.lower() != "none"
-        ):
-            self.navmesh_config_and_recompute()
+        for i in range(self.num_env):
+            if (
+                not self.sim[i].pathfinder.is_loaded
+                and self.cfg.sim_cfg.scene_id.lower() != "none"
+            ):
+                self.navmesh_config_and_recompute(i)
 
         self.time_since_last_simulation = 0.0
         LoggingContext.reinitialize_from_env()
         logger.setLevel("INFO")
         self.print_help_text()
 
-    def draw_contact_debug(self):
+    def draw_contact_debug(self, env_index):
```

Comment:
Is there any way to avoid introducing `env_index` everywhere in viewer.py? E.g. I also see it above for navmesh stuff. I can almost guarantee that nobody will do contact debugging *while* using multiple envs. Using the python viewer with multiple envs is super-niche (basically only Mikael will use this) and so I hate to see `env_index` complicating all the code throughout this file.

As background, when I originally thought about integrating the batch renderer into viewer.py, I was really just thinking about using the batch renderer to render the scene, e.g. so we could troubleshoot visual differences between the renderers. I suspect this tiled viewer use-case is super-niche and something 95% of viewer.py users will never care about.

Reply:
Makes sense. I'll clean that up.